### DISCUSSION TITLE
send request in Theia

### DISCUSSION THREAD

**Discussion by wassimkha** [eclipse-theia/theia#9766] [Q&A] [answered]
Hi,

I'm trying to send a request from the front-end file (`browser` folder). 

```js
const URL = "https://www.google.com/"
fetch(URL).then(function(response) {
        return response.json();
      }).then(function(data) {
        console.log(data);
      }).catch(function(e) {
        console.log("Booo");
        console.log(e)
});
```
I keep getting the error:
```bash
root INFO TypeError: Origin http://localhost:3000 is not allowed by Access-Control-Allow-Origin.
```

How can I send requests from the front end in the theia extension, from the front end files?

**Comment by msujew** [comment, marked as answer]
Hi @wassimkha, looks like you encountered one of the pitfalls of modern web development.

Basically, the server you're sending a request to does not allow any incoming requests from `localhost:3000` (your Theia instance, I assume?) because of [CORS](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing). This is not a Theia issue, but has to be fixed on the server. See [here](https://stackoverflow.com/a/18643011) for more information. What you have to do specifically to make it work, depends on the framework that you use for the server.

**Reply by wassimkha** [reply]
Thank you @msujew for the anwer. Currently, I'm sending a request to `https://www.google.com/` which should allow requests to be sent and received. I'm using that URL for testing, so I don't think that it's a server issue, maybe some parameters on theia need to be modified? 

PS: I updated my description to include the URL

**Reply by msujew** [reply]
Actually, no. The `https://www.google.com` domain does not allow that. Try running your code from the browser console from any domain, and it should show you the same thing (the CORS error). The server has to configure the allowed origins, you can't do anything about that from the 'requesting' side of things.

See also [this example](https://stackoverflow.com/a/56706123) that I found that tries to do the same thing (sending a fetch to google).

**Reply by wassimkha** [reply]
I see you're right, that was the mistake! Looks like other websites are correctly working, thank you!

**Reply by msujew** [reply]
Glad I could help you, you're welcome!
